actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb

```diff
@@ -0,0 +1,77 @@
+require 'active_support/core_ext/object/try'
+
+module ActionView
+  module CollectionCaching # :nodoc:
+    extend ActiveSupport::Concern
+
+    included do
+      # Fallback cache store if Action View is used without Rails.
+      # Otherwise overriden in Railtie to use Rails.cache.
+      mattr_accessor(:collection_cache) { ActiveSupport::Cache::MemoryStore.new }
+    end
+
+    private
+
+    def cache_collection_render
+      return yield unless cache_collection?
+
+      keyed_collection = collection_by_cache_keys
+      partial_cache = collection_cache.read_multi(*keyed_collection.keys)
+
+      @collection = keyed_collection.reject { |key, _| partial_cache.key?(key) }.values
+      rendered_partials = @collection.any? ? yield : []
+
+      fetch_or_cache_partial(partial_cache, order_by: keyed_collection.each_key) do
+        rendered_partials.shift
+      end
+    end
+
+    def cache_collection?
+      if ActionController::Base.perform_caching
+        @options[:cache].present? || automatic_cache_eligible?
+      end
+    end
+
+    def automatic_cache_eligible?
+      if single_template_render? && !callable_cache_key?
+        @template.source =~ /\A<% cache\(?\s*(\w+\.?)/ && $1 == inferred_cache_name.to_s
```

Comment:
Got it! Misremembered turning off capturing for a group as `:?` instead of `?:`. I also added a `\n?`.

Here are my test cases: https://gist.github.com/kaspth/6ba8ac0356549e219b57. Let me know if you find scenarios I've missed. :smile: